streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -411,25 +394,32 @@ public void seek(final TopicPartition partition, final long offset) {
 
         shouldNotSeek.set(new AssertionError("Should not seek"));
 
+        @SuppressWarnings("unchecked")
         final java.util.function.Consumer<Set<TopicPartition>> resetter =
-            EasyMock.mock(java.util.function.Consumer.class);
-        resetter.accept(Collections.singleton(partition1));
-        EasyMock.expectLastCall();
-        EasyMock.replay(resetter);
+            mock(java.util.function.Consumer.class);
```

Comment:
we can just pass lambda function to `task.completeRestoration`. For example:
```java
        // We need to keep a separate reference to the arguments of Consumer#accept
        // because the underlying data-structure is emptied and on verification time
        // it is reported as empty.
        final Set<TopicPartition> partitionsAtCall = new HashSet<>();

        task.initializeIfNeeded();
        task.completeRestoration(partitionsAtCall::addAll);
```

Reply:
Heya @chia7712 and thank you for the review and the good suggestions! Apologies for not responding sooner, I somehow missed it. The newest commits ought to address both of your suggestions 😊